src/exercise/01.js

```diff
@@ -3,16 +3,31 @@
 
 import * as React from 'react'
 
-function Counter({initialCount = 0, step = 1}) {
-  // 🐨 replace React.useState with React.useReducer.
-  // 💰 React.useReducer(countReducer, initialCount)
-  const [count, setCount] = React.useState(initialCount)
+/**
+ * Counter Reducer. Increments the count.
+ * @param state
+ * @param action
+ * @returns {*}
+ */
+const counterReducer = (state, action) => {
+  const {type, step} = action;
+  switch (type) {
+    case 'INCREMENT':
+      return {
+        ...state,
+        count: state.count + step
+      }
+    default:
+      return {}
```

Comment:
I didn't do this in my demo/video, but I realized the default state in a reduce generally should be returning the existing state, so `return { ...state }` instead.